src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs

```diff
@@ -369,9 +369,9 @@ private void OnQueryButtonClick(object sender, RoutedEventArgs e)
 
 		private void SubmitSearch(object item)
 		{
-			var finalResult = item ?? GetObjectText(Text);
+			var finalResult = item;
 
-			QuerySubmitted?.Invoke(this, new AutoSuggestBoxQuerySubmittedEventArgs(finalResult is "" ? null : finalResult, userInput));
+			QuerySubmitted?.Invoke(this, new AutoSuggestBoxQuerySubmittedEventArgs(finalResult, userInput));
```

Comment:
There is no point in assigning `item` to `finalResult`, you can use `item` directly.

Reply:
The `userInput` argument is quite concerning as well. Should it be `Text`? @Xiaoy312 @kazo0

Reply:
@Youssef1313 , In all cases, the event args QueryText property contains the text from the text box.
So in my opinion, userInput is always _textBox.Text

Reply:
> So in my opinion, userInput is always _textBox.Text

Is that *guaranteed*? If yes, why not be on the safe side and use `_textBox.Text` as it's more clear?

Reply:
> > So in my opinion, userInput is always _textBox.Text
> 
> Is that _guaranteed_? If yes, why not be on the safe side and use `_textBox.Text` as it's more clear?

Oh, I was wrong. when we type the text with keyborar to TextBox, userInput refers to _textBox.Text. Otherwise( like prgrammatically change of textbox.text), userInput isn't same with _textBox.Text.

Reply:
I think the second parameter should be _textBox.Text. userInput is wrong.
AutoSuggestBoxQuerySubmittedEventArgs.QueryText is clearly _textBox.Text,
Thank you, @Youssef1313

Reply:
How about giving Text ( i.e. this.Text) as QuerySubmit()?

Reply:
I'd prefer `_textBox.Text` since that is what the [documentation](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.autosuggestbox?view=winrt-22621#query-submitted) explicitly says.

> In all cases, the event args [QueryText](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.autosuggestboxquerysubmittedeventargs.querytext?view=winrt-22621#windows-ui-xaml-controls-autosuggestboxquerysubmittedeventargs-querytext) property contains the text from the text box.

Reply:
@Youssef1313 ,By testing, `this.Text` is wrong! Because this.Text is set after calling the QuerSubmitted handler.

Reply:
https://github.com/unoplatform/uno/blob/9d6de4b11df69c75a2038e31021565971424bf2f/src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs#L349-L358
@Youssef1313 , I think it should be fixed.
And I don't know well if _textBox.Text is right.

Reply:
By calling ChoseItem before SubmitSearch, the this.Text is already set before calling QuerySubmitted.